subgraphs/osmosis/src/modules/Swap.ts

```diff
@@ -0,0 +1,404 @@
+      return;
+    }
+    inputTokenAmounts[tokenOutIndex] = tokenOutAmount;
+  }
+  // log.warning("swap() inputTokenAmounts[0] {} inputTokenAmounts[1] {}", [
+  //   inputTokenAmounts[0].toString(),
+  //   inputTokenAmounts[1].toString(),
+  // ]);
+
+  // if (
+  //   liquidityPool.inputTokenBalances[1] != constants.BIGINT_ZERO &&
+  //   inputTokenBalances[1] != constants.BIGINT_ZERO &&
+  //   inputTokenAmounts[1] != constants.BIGINT_ZERO
+  // ) {
+  //   // if (
+  //   //   inputTokenBalances[0] >= constants.BIGINT_ZERO &&
+  //   //   inputTokenBalances[1] >= constants.BIGINT_ZERO
+  //   // ) {
+
+  //   let price = liquidityPool.inputTokenBalances[0].divDecimal(
+  //     inputTokenBalances[1].toBigDecimal()
+  //   );
+
+  //   if (tokenInIndex == 0) {
+  //     price = inputTokenBalances[0].divDecimal(
+  //       liquidityPool.inputTokenBalances[1].toBigDecimal()
+  //     );
+  //   }
+  //   log.warning(
+  //     "swap for pool {} inputTokenBalances before [0] {} [1] {} ratio {} *******after [0] {} [1] {} ratio {} ****** tokenAmount [0] {} [1] {} ratio {} price {}",
+  //     [
+  //       liquidityPool.id.toString(),
+  //       liquidityPool.inputTokenBalances[0].toString(),
+  //       liquidityPool.inputTokenBalances[1].toString(),
+  //       liquidityPool.inputTokenBalances[0]
+  //         .divDecimal(liquidityPool.inputTokenBalances[1].toBigDecimal())
+  //         .toString(),
+  //       inputTokenBalances[0].toString(),
+  //       inputTokenBalances[1].toString(),
+  //       inputTokenBalances[0]
+  //         .divDecimal(inputTokenBalances[1].toBigDecimal())
+  //         .toString(),
+  //       inputTokenAmounts[0].toString(),
+  //       inputTokenAmounts[1].toString(),
+  //       inputTokenAmounts[0]
+  //         .divDecimal(inputTokenAmounts[1].toBigDecimal())
+  //         .toString(),
+  //       price.toString(),
+  //     ]
+  //   );
+  // }
+
+  liquidityPool.inputTokenBalances = inputTokenBalances;
+  liquidityPool._inputTokenAmounts = inputTokenAmounts;
+  liquidityPool.save();
+
+  let amountInUSD = constants.BIGDECIMAL_ZERO;
+  let amountOutUSD = constants.BIGDECIMAL_ZERO;
+  let volumeUSD = constants.BIGDECIMAL_ZERO;
+  const prevTVL = liquidityPool.totalValueLockedUSD;
+
+  const hasPriceData = utils.updatePoolTVL(liquidityPool, block);
+  if (hasPriceData) {
+    const tokenIn = getOrCreateToken(tokenInDenom);
+    const tokenOut = getOrCreateToken(tokenOutDenom);
+
+    updatePrice(tokenIn, tokenInAmount, tokenOut, tokenOutAmount, block.header);
+
+    if (tokenIn.lastPriceUSD !== null) {
+      amountInUSD = tokenInAmount
+        .divDecimal(
+          constants.BIGINT_TEN.pow(tokenIn.decimals as u8).toBigDecimal()
+        )
+        .times(tokenIn.lastPriceUSD!);
+    }
+    if (tokenOut.lastPriceUSD !== null) {
+      amountOutUSD = tokenOutAmount
+        .divDecimal(
+          constants.BIGINT_TEN.pow(tokenOut.decimals as u8).toBigDecimal()
```

Comment:
Maybe encapsulate this into a function? I have been using convertTokenToDecimal(token.decimal, amount) - your choice though.

Reply:
Fixed at 6903f52ce28319ef61b2eb05dafdabf73a47568c